wpengine/faustjs
# use `?rest_route` query param instead of `/wp-json`

## Description

By default, `/wp-json` is not a defined route. You must first set your permalinks in order to access `/wp-json`.

This leads to invalid behavior when trying to get authentication tokens from Faust WP. 

A request to the token endpoint in the Faust WP plugin responds in a 301 redirect to the node frontend, because the endpoint path doesn't exist.

By using the `?rest_route=/` pattern, we can guarentee the URL will always route to the appropriate API endpoint.



## Related Issue(s):

- #641 

## Testing

1. Create a WordPress site with the FaustWP and WP GraphQL plugins.
2. Set the permalinks to anything but the default
3. Implement the [Local authentication strategy](https://faustjs.org/docs/next/guides/auth#local-based-authentication)
4. Notice that cookies are being saved, tokens are being requested successfully.
5. Now set permalinks to the default
6. Clear cookies
7. Notice now that requests for tokens do not succeed.
8. Now, switch to the `use-query-param-rest-route` branch and repeat steps 5-7 and notice that requests for tokens work as expected.
